src/server/http/__tests__/xsrf.js

```diff
@@ -29,10 +30,31 @@ describe('xsrf request filter', function () {
 
     await kbnServer.ready();
 
-    const routeMethods = nonDestructiveMethods.filter(method => method !== 'HEAD').concat(destructiveMethods);
```

Comment:
__question:__ I had to split route declaration for `GET` and for the rest of methods since `payload` config is only allowed for non-GET/HEAD routes. The payload config is needed to make HAPI happy as in tests it complains about unknown media type (e.g. `application/x-ndjson`). I'm still trying to figure out why it works for the app (e.g. in msearch case) without this config change. 

Do you have any ideas?